File: mental-health-ai/backend/routes/analyze.js
const express = require('express');
const router = express.Router();

router.post('/analyze', async (req, res) => {
  try {
    const { text } = req.body || {};
    if (!text || typeof text !== 'string' || text.trim().length === 0) {
      return res.status(400).json({ error: 'Invalid input: text is required.' });
    }

    const negative = [
      'tired', 'stressed', 'anxious', 'hopeless', 'alone', 'depressed', 'worthless', 'crying', 'insomnia'
    ];
    const critical = ['suicide', 'kill', 'die', 'end it all'];

    const normalized = text.toLowerCase();
    let score = 30; // base
    const signals = [];

    // count occurrences
    negative.forEach((kw) => {
      const re = new RegExp('\\b' + kw.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&') + '\\b', 'g');
      const matches = normalized.match(re);
      if (matches && matches.length) {
        score += 10 * matches.length;
        signals.push(...Array.from(new Set([...(signals || []), kw])));
      }
    });

    critical.forEach((kw) => {
      const re = new RegExp('\\b' + kw.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&') + '\\b', 'g');
      const matches = normalized.match(re);
      if (matches && matches.length) {
        score += 20 * matches.length;
        signals.push(...Array.from(new Set([...(signals || []), kw])));
      }
    });

    if (score > 100) score = 100;

    let message = 'Low to moderate risk.';
    if (score >= 70) message = 'High risk — immediate attention recommended.';
    else if (score >= 40) message = 'Moderate risk — monitor and consider support.';

    return res.json({ riskScore: score, signals: Array.from(new Set(signals)), message });
  } catch (err) {
    console.error('Analyze error', err);
    return res.status(500).json({ error: 'Server error' });
  }
});

module.exports = router;

File: mental-health-ai/backend/routes/analyze.py
from flask import Blueprint, request, jsonify
import re

analyze_bp = Blueprint('analyze', __name__)

NEGATIVE_KEYWORDS = [
    'tired', 'stressed', 'anxious', 'hopeless', 'alone', 'depressed', 'worthless', 'crying', 'insomnia'
]

CRITICAL_KEYWORDS = ['suicide', 'kill', 'die', 'end it all']

@analyze_bp.route('/analyze', methods=['POST'])
def analyze():
    try:
        data = request.get_json()
        text = data.get('text') if data else None
        
        if not text or not isinstance(text, str) or len(text.strip()) == 0:
            return jsonify({'error': 'Invalid input: text is required.'}), 400
        
        normalized = text.lower()
        score = 30  # base score
        signals = []
        
        # Check for negative keywords
        for kw in NEGATIVE_KEYWORDS:
            pattern = r'\b' + re.escape(kw) + r'\b'
            matches = re.findall(pattern, normalized)
            if matches:
                score += 10 * len(matches)
                signals.append(kw)
        
        # Check for critical keywords
        for kw in CRITICAL_KEYWORDS:
            pattern = r'\b' + re.escape(kw) + r'\b'
            matches = re.findall(pattern, normalized)
            if matches:
                score += 20 * len(matches)
                signals.append(kw)
        
        # Cap at 100
        if score > 100:
            score = 100
        
        # Remove duplicates
        signals = list(set(signals))
        
        # Determine message
        if score >= 70:
            message = 'High risk — immediate attention recommended.'
        elif score >= 40:
            message = 'Moderate risk — monitor and consider support.'
        else:
            message = 'Low to moderate risk.'
        
        return jsonify({
            'riskScore': score,
            'signals': signals,
            'message': message
        }), 200
        
    except Exception as err:
        print('Analyze error:', err)
        return jsonify({'error': 'Server error'}), 500
